### PR TITLE
new user functions utils

### PR DESCRIPTION
**Pull request by danielhaap83**
Implements https://github.com/keirlawson/rusty-git/issues/9

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -156,6 +156,26 @@ impl Repository {
         };
         execute_git_fn(&self.location, args, |output| output.trim().to_owned())
     }
+
+    /// Execute user defined command
+    pub fn utils<I, S>(&self, args: I) -> Result<()>
```
Comment (keirlawson) [inline src/lib.rs]: not sure this is a great name for such a function, I think something along the lines of `command` would communicate intent more clearly

File: src/lib.rs [hunk 2]
```diff
@@ -156,6 +156,26 @@ impl Repository {
...
+    where
+        I: IntoIterator<Item = S>,
+        S: AsRef<OsStr>,
+    {
+        execute_git(&self.location, args)
+    }
+
+    /// Execute user defined command with fun
```
Comment (keirlawson) [inline src/lib.rs]: don't think its immediately obvious what `fun` is in this context

Comment (danielhaap83) [reply src/lib.rs]: OK. What do you suggest to differ between running a command with and without output from console? I know this kind of naming from https://docs.rs/cmd_lib_core/0.4.0/cmd_lib_core/.

How about renaming:
- utils -> cmd
- utils_fun -> cmd_out

Comment (keirlawson) [reply src/lib.rs]: Apologies for my slowness in getting back on this, yep those names lgtm :)

Comment (danielhaap83) [reply src/lib.rs]: Done

**Review by keirlawson — APPROVED** [review]

**Pull request merged**